core/src/main/java/org/mule/runtime/core/internal/message/ErrorBuilder.java

```diff
@@ -65,14 +69,36 @@ private ErrorBuilder(Throwable e) {
     if (e instanceof ComposedErrorException) {
       this.errors = ((ComposedErrorException) e).getErrors();
     }
+    if (e instanceof SuppressedMuleException) {
+      this.suppressedErrors = getSuppressedMessagingErrors((SuppressedMuleException) e);
+    }
     this.exception = e;
     String exceptionDescription = e.getMessage() != null ? e.getMessage() : "unknown description";
     this.description = exceptionDescription;
     this.detailedDescription = exceptionDescription;
     MuleException muleRoot = getRootMuleException(this.exception);
-    if (muleRoot != null && muleRoot.getMessage() != null) {
-      this.description = muleRoot.getMessage();
+    if (muleRoot != null) {
+      this.suppressedErrors = getSuppressedMessagingErrors(muleRoot);
+      if (muleRoot.getMessage() != null) {
+        this.description = muleRoot.getMessage();
+      }
+    }
+  }
+
+  /**
+   * Returns all the {@link Error} instances that the given {@link MuleException} inform as suppressed.
+   * @param muleException Given {@link MuleException}.
+   * @return {@link List<Error>} containing all the {@link Error} instances that the given {@link MuleException} inform as suppressed.
+   * @see SuppressedMuleException
+   */
+  private List<Error> getSuppressedMessagingErrors(MuleException muleException) {
```

Comment:
the caller is casting the parameter to SuppressedMuleException, why is this receiving a more generic parameter?

Reply:
Note that there are two calls to this method, the one that you are talking about and a second one where the parameter is a MuleException (the root mule exception, if present)